iModelCore/ECPresentation/PublicAPI/ECPresentation/NavNodeKey.h

```diff
@@ -507,4 +507,18 @@ struct NavNodeKeyListContainer : INavNodeKeysContainer
 typedef bvector<ECInstanceKey> GroupedInstanceKeysList;
 typedef GroupedInstanceKeysList const& GroupedInstanceKeysListCR;
 
+/*=================================================================================**//**
+* @bsiclass
++===============+===============+===============+===============+===============+======*/
+struct INodeInstanceKeysProvider
+    {
+    protected:
+        virtual void _IterateInstanceKeys(NavNodeKeyCR, std::function<bool(ECInstanceKey)>) const = 0;
+        virtual bool _ContainsInstanceKey(NavNodeCR, ECInstanceKeyCR) const = 0;
+    public:
+        virtual ~INodeInstanceKeysProvider() {}
+        void IterateInstanceKeys(NavNodeKeyCR nodeKey, std::function<bool(ECInstanceKey)> cb) const { _IterateInstanceKeys(nodeKey, cb); }
+        bool ContainsInstanceKey(NavNodeCR node, ECInstanceKeyCR instanceKey) const { return _ContainsInstanceKey(node, instanceKey); }
+    };

```

Comment:
- Formatting - when a class/struct has access modifiers, we don't indent curly braces and modifiers to avoid double indent on members
- Now that the interface is public, we should attempt to make the API as good as possible. I believe `ContainsInstanceKey` could take `NavNodeKeyCR` instead of `NavNodeCR`